core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -283,12 +283,12 @@ protected JSONSignatureValidator getJsonSignatureValidator(@CheckForNull String
     /**
      * Returns true if it's time for us to check for new version.
      */
-    public boolean isDue() {
+    synchronized public boolean isDue() {
```

Comment:
```suggestion
    public synchronized boolean isDue() {
```

Typical keyword ordering.